commonutils/src/main/java/com/asgatech/commonutils/permissions/PermissionHandler.kt

```diff
@@ -0,0 +1,155 @@
+package com.asgatech.commonutils.permissions
+
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+
+
+/**
+ * @usage: handle the checking of granted permission and to make it easier to handle asking for not granted permissions
+ */
+class PermissionHandler {
+    private val PERMISSION_REQ = 452
+    private var permissionCallback: PermissionsCallback? = null
+
+    /**
+     * Check if certain permission is granted
+     */
+    fun isPermissionGranted(context: Context, permission: String): Boolean {
+        val selfPermission = ContextCompat.checkSelfPermission(context, permission)
+        return selfPermission == PackageManager.PERMISSION_GRANTED
+    }
+
+    /**
+     * Check if certain permissions are granted or not.
+     */
+    fun isPermissionGranted(context: Context, permissions: Array<String>): Boolean {
+        val grantedOnes: MutableList<String> = ArrayList()
+        for (permission in permissions) {
```

Comment:
you can optimize the performance by check if isPErmissionGranted == false then return false

and replace grantedOnes.size == permissions.size with true